src/engine/Timer.ts

```diff
@@ -1,5 +1,11 @@
 import { Scene } from './Scene';
 
+export interface TimerOptions {
```

Comment:
I like where you are going with this, we have come up with similar ideas for refactoring the constructor 👍 

https://github.com/excaliburjs/Excalibur/issues/396#issuecomment-538583725